staging/src/k8s.io/csi-translation-lib/plugins/gce_pd.go

```diff
@@ -286,7 +291,10 @@ func (g *gcePersistentDiskCSITranslator) TranslateCSIPVToInTree(pv *v1.Persisten
 		gceSource.Partition = int32(partInt)
 	}
 
-	// TODO: Take the zone/regional information and stick it into the label.
+	// translate CSI topology to In-tree topology for rollback compatibility
+	if err := translateTopologyFromCSIToInTree(pv, GCEPDTopologyKey); err != nil {
+		return nil, fmt.Errorf("failed to translate topology: %v", err)
+	}
 
 	pv.Spec.CSI = nil
 	pv.Spec.GCEPersistentDisk = gceSource
```

Comment:
I think we also have to enhance the storageclass translation to handle AllowedTopologies having beta/ga labels

Reply:
nm i saw this was done in the common library